Sources/TensorFlow/Layers/Normalization.swift

```diff
@@ -98,18 +98,19 @@ public struct BatchNorm<Scalar: TensorFlowFloatingPoint>: Layer {
   ///
   /// - Parameter input: The input to the layer.
   /// - Returns: The output.
-  @differentiable
-  public func forward(_ input: Tensor<Scalar>) -> Tensor<Scalar> {
-    let positiveAxis = (input.rank + axis) % input.rank
+  @differentiable(reverse)
+  public func callAsFunction(_ input: Tensor<Scalar>) -> Tensor<Scalar> {
+    let inputRank = input.rank
+    let positiveAxis = (inputRank + axis) % inputRank
```

Comment:
Wouldn't this suggestion harm performance because you have to call `input.rank` twice? That would increase the total overhead from 3/4 to 6/8 function calls.